### PR TITLE
torchtext SentencePiece with torch.utils.data.DataLoader

### PR DESCRIPTION
**Pull request by zhangguanheng66**
Add a CI test to check the compatibility between torchtext SentencePiece and `torch.utils.data.DataLoader`.

### CODE REVIEW

File: test/experimental/test_transforms_with_asset.py [hunk 1]
```diff
@@ -169,6 +170,21 @@ def test_builtin_pretrained_sentencepiece_processor(self):
...
+        spm_processor = sentencepiece_processor(sp_model_path)
+        _path = os.path.join(self.project_root, '.data', 'text_bpe_25000.model')
+        os.remove(_path)
+        example_strings = ['the pretrained spm model names'] * 64
+        ref_results = torch.tensor([[13, 1465, 12824, 304, 24935, 5771, 3776]] * 16, dtype=torch.long)
+
+        def batch_func(data):
+            return torch.tensor([spm_processor(text) for text in data], dtype=torch.long)
+
+        dataloader = DataLoader(example_strings, collate_fn=batch_func, batch_size=16)
```
Comment (cpuhrsch) [inline test/experimental/test_transforms_with_asset.py:184]: [By default DataLoader doesn't do multiprocessing](https://pytorch.org/docs/master/data.html?highlight=dataloader#torch.utils.data.DataLoader). Set the number of workers to a non-zero value (1 is enough to exercise serialization).

**Comment by zhangguanheng66** [comment]
Open an issue to track the problem with the Windows platform. https://github.com/pytorch/text/issues/1076

File: test/experimental/test_transforms_with_asset.py [hunk 2]
```diff
@@ -169,6 +176,18 @@ def test_builtin_pretrained_sentencepiece_processor(self):
...
 
+    def test_sentencepiece_with_dataloader(self):
+        sp_model_path = download_from_url(PRETRAINED_SP_MODEL['text_bpe_25000'])
+        spm_processor = sentencepiece_processor(sp_model_path)
+        _path = os.path.join(self.project_root, '.data', 'text_bpe_25000.model')
+        os.remove(_path)
+        example_strings = ['the pretrained spm model names'] * 64
+        ref_results = torch.tensor([[13, 1465, 12824, 304, 24935, 5771, 3776]] * 16, dtype=torch.long)
+        dataloader = DataLoader(example_strings, batch_size=16, num_workers=2,
+                                collate_fn=lambda data_: batch_func(data_, spm_processor))
```
Comment (peterjc123) [inline test/experimental/test_transforms_with_asset.py:187]: Same for collate_fn.

Comment (peterjc123) [reply test/experimental/test_transforms_with_asset.py:187]: lambda functions won't work too.

**Comment by peterjc123** [comment]
Please add `certifi` to https://github.com/pytorch/text/blob/master/.circleci/unittest/windows/scripts/environment.yml#L7.

**Comment by zhangguanheng66** [comment]
> Please add `certifi` to https://github.com/pytorch/text/blob/master/.circleci/unittest/windows/scripts/environment.yml#L7.

Yup. That's a separate issue, and a PR is welcome.

File: test/experimental/test_transforms_with_asset.py [hunk 3]
```diff
@@ -169,6 +181,17 @@ def test_builtin_pretrained_sentencepiece_processor(self):
         ref_results = [13, 1465, 12824, 304, 24935, 5771, 3776]
         self.assertEqual(spm_transform(test_sample), ref_results)
 
+    # we separate out these errors because Windows runs into seg faults when propagating
+    # exceptions from C++ using pybind11
+    @unittest.skipIf(platform.system() == "Windows", "Test is known to fail on Windows.")
```
Comment (cpuhrsch) [inline test/experimental/test_transforms_with_asset.py:186]: If we're disabling it for windows we can probably move the functions back into the text

**Review by cpuhrsch — APPROVED** [review]

**Pull request merged**